src/modules/EmailRecoveryModule.sol

```diff
@@ -0,0 +1,175 @@
+import { IERC7579Account } from "erc7579/interfaces/IERC7579Account.sol";
+import { IModule } from "erc7579/interfaces/IERC7579Module.sol";
+import { ExecutionLib } from "erc7579/lib/ExecutionLib.sol";
+import { ModeLib } from "erc7579/lib/ModeLib.sol";
+
+import { IRecoveryModule } from "../interfaces/IRecoveryModule.sol";
+import { IEmailRecoveryManager } from "../interfaces/IEmailRecoveryManager.sol";
+import { ISafe } from "../interfaces/ISafe.sol";
+import { BytesLib } from "../libraries/BytesLib.sol";
+
+contract EmailRecoveryModule is ERC7579ExecutorBase, IRecoveryModule {
+    using BytesLib for bytes;
+    /*//////////////////////////////////////////////////////////////////////////
+                                    CONSTANTS
+    //////////////////////////////////////////////////////////////////////////*/
+
+    address public immutable emailRecoveryManager;
+
+    event NewValidatorRecovery(address indexed validatorModule, bytes4 recoverySelector);
+
+    error NotTrustedRecoveryManager();
+    error InvalidSubjectParams();
+    error InvalidValidator(address validator);
+    error InvalidSelector(bytes4 selector);
+
+    mapping(address validatorModule => mapping(address account => bytes4 allowedSelector)) internal
+        allowedSelectors;
+    mapping(address account => address validator) internal validators;
+
+    constructor(address _zkEmailRecovery) {
+        emailRecoveryManager = _zkEmailRecovery;
+    }
+
+    modifier withoutUnsafeSelector(bytes4 recoverySelector) {
+        if (
+            recoverySelector == IModule.onUninstall.selector
+                || recoverySelector == IModule.onInstall.selector
+        ) {
+            revert InvalidSelector(recoverySelector);
+        }
+
+        _;
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                     CONFIG
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /**
+     * Initialize the module with the given data
+     * @param data The data to initialize the module with
+     */
+    function onInstall(bytes calldata data) external {
+        (
+            address validator,
+            bytes4 selector,
+            address[] memory guardians,
+            uint256[] memory weights,
+            uint256 threshold,
+            uint256 delay,
+            uint256 expiry
+        ) = abi.decode(data, (address, bytes4, address[], uint256[], uint256, uint256, uint256));
+
+        allowValidatorRecovery(validator, bytes("0"), selector);
+        validators[msg.sender] = validator;
+
+        _execute({
+            to: emailRecoveryManager,
+            value: 0,
+            data: abi.encodeCall(
+                IEmailRecoveryManager.configureRecovery,
+                (address(this), guardians, weights, threshold, delay, expiry)
+            )
+        });
+    }
+
+    function allowValidatorRecovery(
```

Comment:
it would be good to add a set/remove function for the recovery allowedSelectors after installation as well

Reply:
Agree will add that